common/catalog-client/src/test/java/com/csgroup/s2pdgs/catalog/adapter/elasticsearch/ElasticsearchAssetCatalogClientManagerTest.java

```diff
@@ -0,0 +1,56 @@
+package com.csgroup.s2pdgs.catalog.adapter.elasticsearch;
+
+import com.csgroup.s2pdgs.catalog.Catalog;
+import com.csgroup.s2pdgs.catalog.ElasticsearchCatalog;
+import com.csgroup.s2pdgs.catalog.adapter.elasticsearch.error.NewClientNeededException;
+import com.csgroup.s2pdgs.catalog.adapter.elasticsearch.json.ElasticsearchJsonCodec;
+import com.csgroup.s2pdgs.catalog.adapter.elasticsearch.json.ElasticsearchObjectMapperBuilder;
+import org.junit.Test;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.*;
+
+public class ElasticsearchAssetCatalogClientManagerTest {
+
+    public static final ElasticsearchJsonCodec CODEC = new ElasticsearchJsonCodec(new ElasticsearchObjectMapperBuilder().mapper());
+    public static final String TEST_INDEX = "test-index";
+
+    static ElasticsearchAssetCatalogClient failingClient = mock(ElasticsearchAssetCatalogClient.class);
+    static ElasticsearchAssetCatalogClient okClient = mock(ElasticsearchAssetCatalogClient.class);
+    static ElasticsearchCatalog catalog = mock(ElasticsearchCatalog.class);
+
+    @Test
+    public void testRecreateClientOnNewClientNeededException () {
+        // GIVEN
+
+        when(failingClient.createIndex(TEST_INDEX)).thenAnswer(i -> Mono.error(new NewClientNeededException(new RuntimeException())));
+        when(okClient.createIndex(TEST_INDEX)).thenAnswer(i -> Mono.just(true));
+
+        AtomicInteger callCount = new AtomicInteger();
+        ElasticsearchAssetCatalogClientManager.ClientFactory testFactory = c -> {
+            if (callCount.getAndIncrement() == 0) { return failingClient; }
+            else { return okClient; }
+        };
+        ElasticsearchAssetCatalogClientManager manager = new ElasticsearchAssetCatalogClientManager(testFactory);
+
+        // WHEN
+        boolean result = manager.monoWithCatalogClient(
+            catalog,
+            client -> client.createIndex(TEST_INDEX)
+        )
+        .subscribeOn(Schedulers.immediate())
+        .block();
+
+        //THEN
+        assertThat(callCount).hasValue(2);
+        assertThat(result).isTrue();
+        verify(failingClient, times(1)).createIndex(TEST_INDEX);
```

Comment:
Minor: argument `times(1)` can be omitted (default value)
```suggestion
        verify(failingClient).createIndex(TEST_INDEX);
```